code/modules/economy/economy_machinery/eftpos.dm

```diff
@@ -218,3 +219,91 @@
 	R.forceMove(D)
 	D.wrapped = R
 	D.name = "small parcel - 'EFTPOS access code'"
+
+/obj/item/eftpos/proc/check_user_position(mob/user)
+	return Adjacent(user)
+
+/obj/item/eftpos/register
+	name = "point of sale"
+	desc = "Also known as a cash register, or, more commonly, \"robbery magnet\". It's old and rusty, and had an EFTPOS module fitted in it."
+	icon = 'icons/obj/machines/pos.dmi'
+	icon_state = "pos"
+	force = 10
+	throwforce = 10
+	throw_speed = 1.5
+	throw_range = 7
+	anchored = TRUE
+	w_class = WEIGHT_CLASS_BULKY
+	hitsound = 'sound/weapons/ringslam.ogg'
+	drop_sound = 'sound/items/handling/register_drop.ogg'
+	pickup_sound =  'sound/items/handling/toolbox_pickup.ogg'
+	transaction_sound = 'sound/machines/checkout.ogg'
+	attack_verb = list("bounced a check off", "checked-out", "tipped")
+
+/obj/item/eftpos/register/examine(mob/user)
+	. = ..()
+	if(!anchored)
+		. += "<span class='notice'>Alt-click to rotate it.</span>"
+	else
+		. += "<span class='notice'>It is secured in place.</span>"
+
+/obj/item/eftpos/register/AltClick(mob/user)
+	if(user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!Adjacent(user))
+		return
+	if(anchored)
+		to_chat(user, "<span class='warning'>[src] is secured in place!</span>")
+		return
+	setDir(turn(dir, 90))
+
+/obj/item/eftpos/register/attack_hand(mob/user)
+	if(anchored)
+		if(!check_user_position(user))
+			to_chat(user, "<span class='warning'>You need to be behind \the [src] to use it!</span>")
+			return
+		add_fingerprint(user)
+		ui_interact(user, state = GLOB.human_adjacent_state)
+		return TRUE
+	return ..()
+
+/obj/item/eftpos/register/attack_self(mob/user)
+	to_chat(user, "<span class='notice'>\The [src] has to be set down and secured to be used.</span>")
+
+/obj/item/eftpos/register/check_user_position(mob/user)
+	if(!..())
+		return FALSE
+	var/user_loc = get_dir(src, user)
+	if(!user_loc || dir == user_loc) // User is sitting right on top of it or standing directly behind it, don't check the corners
+		return TRUE
+	var/valid_loc = FALSE
+	switch(dir)
+		if(NORTH)
+			if(user_loc == NORTHEAST || user_loc == NORTHWEST)
+				valid_loc = TRUE
+		if(SOUTH)
+			if(user_loc == SOUTHEAST || user_loc == SOUTHWEST)
+				valid_loc = TRUE
+		if(EAST)
+			if(user_loc == NORTHEAST || user_loc == SOUTHEAST)
+				valid_loc = TRUE
+		if(WEST)
+			if(user_loc == NORTHWEST || user_loc == SOUTHWEST)
+				valid_loc = TRUE
+	return valid_loc
```

Comment:
There's actually shortcuts we can use here with bitwise operations
```dm
	if(!..())
		return FALSE
	var/user_loc = get_dir(src, user)
	if(!user_loc || user_loc & dir)
		return TRUE
	return FALSE
```
this is because directions are stored as bitshifts
We know the following
```
NORTH = 1		0001
NORTHEAST = 5 	0101
NORTHWEST = 9	1001

All of these "face north" at least partially and have the NORTH bit active, so we just need to check that bit

NORTHWEST	| 1 0 0 1												
NORTH		| 0 0 0 1 &											
			-----------									
			 0 0 0 1 --> This number is TRUE		

However, you will see for none North dirs it returns 0 or FALSE	

SOUTHWEST	| 1 0 1 0
NORTH		| 0 0 0 1 &
			-----------
			 0 0 0 0 --> FALSE
```